File: Creacion API/APIRest2_23_24_Capas/APIRest2_23_24/src/main/java/es/luisherrero/apirest1/model/Comentario.java
package es.luisherrero.apirest1.model;

import java.io.Serializable;
import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@jakarta.persistence.Entity
@Table(name="comentarios")
public class Comentario implements Serializable{
	private static final long serialVersionUID=1;

	@Id
	private int id;

	private String adjuntoUrl;

	private Date fechahora;

	private String texto;

	@Column(name="incidencia_num")
	private Incidencia incidencia;

	@Column(name="personal_id")
	private Personal personal;

	public Comentario() {
	}

	public int getId() {
		return this.id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getAdjuntoUrl() {
		return this.adjuntoUrl;
	}

	public void setAdjuntoUrl(String adjuntoUrl) {
		this.adjuntoUrl = adjuntoUrl;
	}

	public Date getFechahora() {
		return this.fechahora;
	}

	public void setFechahora(Date fechahora) {
		this.fechahora = fechahora;
	}

	public String getTexto() {
		return this.texto;
	}

	public void setTexto(String texto) {
		this.texto = texto;
	}

	public Incidencia getIncidencia() {
		return this.incidencia;
	}

	public void setIncidencia(Incidencia incidencia) {
		this.incidencia = incidencia;
	}

	public Personal getPersonal() {
		return this.personal;
	}

	public void setPersonal(Personal personal) {
		this.personal = personal;
	}
}

File: Creacion API/Python/comentario.py
class Comentario:
    def __init__(self, id, texto, fechahora, incidencia_num, personal_id, adjunto_url):
        self.id = id
        self.texto = texto
        self.fechahora = fechahora
        self.incidencia_num = incidencia_num
        self.personal_id = personal_id
        self.adjunto_url = adjunto_url

    def get_id(self):
        return self.id

    def set_id(self, value):
        self.id = value
        
    def get_texto(self):
        return self.texto

    def set_texto(self, value):
        self.texto = value
    
    def get_fechahora(self):
        return self.fechahora

    def set_fechahora(self, value):
        self.fechahora = value
    
    def get_incidencia_num(self):
        return self.incidencia_num

    def set_incidencia_num(self, value):
        self.incidencia_num = value
        
    def get_personal_id(self):
        return self.personal_id

    def set_personal_id(self, value):
        self.personal_id = value    
        
    def get_adjunto_url(self):
        return self.adjunto_url

    def set_adjunto_url(self, value):
        self.adjunto_url = value